medkit/text/deid.py

```diff
@@ -35,3 +37,113 @@ def _run_one(self, segment: Segment) -> Iterator[Entity]:
                 spans=spans,
                 metadata={"score": result.score},
             )
+
+
+class FrDateRecognizer(PatternRecognizer):
+    """
+    Recognizes French Date using regex.
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    """
+
+    PATTERNS: ClassVar[list[Pattern]] = [
+        Pattern(
+            "French dates with day month year",
+            r"\b(?i)(\d{1,2}|1er) "
+            r"((janvier|janv.|JAN)|(février|févr.|FÉV)|(mars|MAR)|"
+            r"(avril|avr.|AVR)|(mai|MAI)|(juin|JUN)|(juillet|juill.|JUL)|"
+            r"(août|AOÛ)|(septembre|sept.|SEP)|(octobre|oct.|OCT)|"
+            r"(novembre|nov.|NOV)|(décembre|déc|DÉC)) \d{4}\b",
+            0.9,
+        ),
+        Pattern(
+            "French dates with month year",
+            r"\b(?i)((janvier|janv.|JAN)|(février|févr.|FÉV)|(mars|MAR)|"
+            r"(avril|avr.|AVR)|(mai|MAI)|(juin|JUN)|(juillet|juill.|JUL)|"
+            r"(août|AOÛ)|(septembre|sept.|SEP)|(octobre|oct.|OCT)|"
+            r"(novembre|nov.|NOV)|(décembre|déc|DÉC)) \d{4}\b",
+            0.9,
+        ),
+        Pattern(
+            "French dates with day month",
+            r"\b(?i)(\d{1,2} |1er )"
+            r"((janvier|janv.|JAN)|(février|févr.|FÉV)|(mars|MAR)|"
+            r"(avril|avr.|AVR)|(mai|MAI)|(juin|JUN)|(juillet|juill.|JUL)|"
+            r"(août|AOÛ)|(septembre|sept.|SEP)|(octobre|oct.|OCT)|"
+            r"(novembre|nov.|NOV)|(décembre|déc|DÉC))\b",
+            0.9,
+        ),
+        Pattern(
+            "French dates with month",
+            r"\b(?i)((janvier|janv.|JAN)|(février|févr.|FÉV)|(mars|MAR)|"
+            r"(avril|avr.|AVR)|(mai|MAI)|(juin|JUN)|(juillet|juill.|JUL)|"
+            r"(août|AOÛ)|(septembre|sept.|SEP)|(octobre|oct.|OCT)|"
+            r"(novembre|nov.|NOV)|(décembre|déc|DÉC))\b",
+            0.9,
+        ),
+    ]
+
+    def __init__(
+        self,
+        patterns: list[Pattern] | None = None,
+        supported_language: str = "fr",
+        supported_entity: str = "FR_DATE",
+    ):
+        patterns = patterns if patterns else self.PATTERNS
+        super().__init__(
+            supported_entity=supported_entity,
+            patterns=patterns,
+            supported_language=supported_language,
+        )
+
+
+class FrAgeRecognizer(PatternRecognizer):
+    """
+    Recognizes French Phone Age using regex.
+
+    :param patterns: List of patterns to be used by this recognizer
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
```

Comment:
To be updated based on my previous comment above

Reply:
Same comment as above

Reply:
I tested it, and this line is necessary, but thank you for the advice